src/arrayMethodSort.js

```diff
@@ -4,8 +4,17 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = (a, b) =>
+    String(a) > String(b) ? 1 : -1) {
+    for (let index = 1; index < this.length; index++) {
```

Comment:
If I understand correctly, your loop has a fixed amount of iterations. By using do or while you can develop functionality when it will work till some changes are made.